nix/tests.nix

```diff
@@ -143,6 +144,58 @@ let
             postgrest --dump-schema "$rootdir"/test/io-tests/configs/simple.config \
             | ${yq}/bin/yq -y .
       '';
+
+  coverage =
+    name: postgresql:
+    let
+      covCabalOptions = "${devCabalOptions} --enable-coverage --enable-executable-dynamic";
+    in
+    checkedShellScript
+      name
+      ''
+        env="$(cat ${postgrest.env})"
+        export PATH="$env/bin:$PATH"
+
+        rootdir="$(${git}/bin/git rev-parse --show-toplevel)"
+        cd "$rootdir"
+
+        # clean up previous coverage data
+        rm -rf coverage
+        find . -type f -iname "*.tix" -print0 | xargs -0 rm -f
+
+        # build once before running all the tests
+        ${cabal-install}/bin/cabal v2-build ${covCabalOptions} --enable-tests all
+
+        ${withTmpDb postgresql} ${cabal-install}/bin/cabal v2-exec ${covCabalOptions} \
+          ${coveragePathHelper} -- ${ioTestPython}/bin/pytest -v test/io-tests
+        ${withTmpDb postgresql} ${cabal-install}/bin/cabal v2-test ${covCabalOptions}
+
+        # collect all the hpc files in one location
+        hpcdir="$(find ./dist-newstyle -type d -iname dyn)"
+        mkdir -p coverage
+        cp -r "$hpcdir"/mix/postgrest*/* coverage
+        hpc sum --output=coverage/all.tix --union --exclude=Paths_postgrest postgrest.tix "$hpcdir"/tix/postgrest*/*.tix
```

Comment:
These (hpc) should also come from some nix derivation.

Reply:
Ah, yes, thanks for bringing that up again. It's in ghc itself and I couldn't figure out how to get it properly at the time. Now I get it from `pkgs.haskell.compiler."${compiler}"` - that works.